frontend/src/metabase/search/components/SearchResult/components/ItemIcon.tsx

```diff
@@ -0,0 +1,32 @@
+import type { SearchModelType } from "metabase-types/api";
+import { IconWrapper } from "metabase/search/components/SearchResult/SearchResult.styled";
+import type { WrappedResult } from "metabase/search/types";
+import { CollectionIcon } from "./CollectionIcon";
+import { DefaultIcon } from "./DefaultIcon";
+import { TableIcon } from "./TableIcon";
+
+const ModelIconComponentMap = {
+  table: TableIcon,
+  collection: CollectionIcon,
+};
+
+export function ItemIcon({
+  item,
+  type,
+  active,
+}: {
+  item: WrappedResult;
+  type: SearchModelType;
+  active: boolean;
+}) {
+  const IconComponent =
+    type in Object.keys(ModelIconComponentMap)
+      ? ModelIconComponentMap[type as keyof typeof ModelIconComponentMap]
+      : DefaultIcon;
+
+  return (
+    <IconWrapper item={item} type={type} active={active}>
+      <IconComponent item={item} />
+    </IconWrapper>
+  );
+}
```

Comment:
WDYT about this?

```suggestion
import type { SearchModelType } from "metabase-types/api";
import { Icon } from "metabase/core/components/Icon";
import type { WrappedResult } from "metabase/search/types";
import { CollectionIcon } from "./CollectionIcon";
import { DefaultIcon } from "./DefaultIcon";
import { IconWrapper } from "./ItemIcon.styled";

interface IconComponentProps {
  item: WrappedResult;
  type: SearchModelType;
}

const IconComponent = ({ item, type }: IconComponentProps) => {
  if (type === "table") {
    return <Icon name="database" />;
  }

  if (type === "collection") {
    return <CollectionIcon item={item} />;
  }

  return <DefaultIcon item={item} />;
};

interface ItemIconProps {
  active: boolean;
  item: WrappedResult;
  type: SearchModelType;
}

export function ItemIcon({ active, item, type }: ItemIconProps) {
  return (
    <IconWrapper type={type} active={active}>
      <IconComponent item={item} type={type} />
    </IconWrapper>
  );
}
```

This way we can:
- get rid of `TableIcon` component
- make sure `item` prop is passed only to the components that need it (previously `TableIcon` received it)
- avoid creating a "framework" for the icons